src/org/opendatakit/briefcase/ui/export/components/ConfigurationPanelForm.java

```diff
@@ -292,13 +293,20 @@ private void setPullBefore(OverridableBoolean value) {
   }
 
   private void setOverwriteFiles(boolean enabled) {
-    overwriteFiles = overwriteFiles.set(enabled);
-    triggerOnChange();
+    if (!enabled || confirmOverwriteFiles()) {
+      overwriteFiles = overwriteFiles.set(enabled);
+      triggerOnChange();
+    } else
+      overwriteFilesField.setSelected(false);
   }
 
   private void setOverwriteFiles(TriStateBoolean overrideValue) {
-    overwriteFiles = overwriteFiles.overrideWith(overrideValue);
-    triggerOnChange();
+    TriStateBoolean previousValue = overwriteFiles.getOverride();
+    if (previousValue != overrideValue && (overrideValue != TRUE || confirmOverwriteFiles())) {
```

Comment:
Not a big deal, but the all-caps TRUE threw me off until I found out what it is. No need to change it for me, but alternatives that occur to me:
overrideValue != TriStateBoolean.TRUE
! isTrue(overrideValue)

Reply:
Thanks, Dave!

Maybe we could think of another enum name to use instead of `TRUE`?